### ISSUE TITLE
The `preset` `satellite` component does not provide satellite orbital position/velocity

### ISSUE THREAD

**Issue by abensonca** [galacticusorg/galacticus#504]
This is problematic as it means that, e.g. `hotHaloRamPressureForceOrbitalPosition` can not work with this component as it requires the instantaneous position and velocity of a satellite.

A solution would be to make the `preset` `satellite` component simply (and, perhaps, optionally) compute position and velocity from the positions of the node and its host. In the typical application of the `preset` `satellite` component - N-body merger trees - we likely also have preset positions, so this would be possible.

**Comment by abensonca** [comment]
To make this most useful, the `preset` `position` component should then interpolate positions/velocities between timesteps. This could be done using the `nodeOperatorPositionInterpolated` class.

**Comment by abensonca** [comment]
## Analysis of the proposed solutions

I went through the current state of the code with this issue in mind. Short version: **the follow-up comment's proposal (interpolated positions) is already done**, and **the ram-pressure symptom that motivated the issue was worked around rather than fixed**, so the issue as filed is still open but its scope has shifted. The remaining value in the original proposal is large, but not for ram pressure — it is for the ~26 *other* classes that need satellite orbital position/velocity and are still unusable with N-body trees.

### State of play

**1. The interpolation half is complete.** `nodeOperatorPositionInterpolated` (`source/nodes/operators/physics/position/interpolated.F90`) precomputes Merson et al. (2013) trajectory coefficients per branch tip in `nodeTreeInitialize`, marks `position`/`velocity` analytic, and evaluates them in `differentialEvolutionSolveAnalytics`. That is called from inside the ODE right-hand side (`source/merger_trees/node_evolver/standard.F90:884`), so positions and velocities are continuous and available at *every* ODE substep, and the adaptive integrator will resolve rapid variation in anything that depends on them. The interpolating getters that used to live in the `cartesian` position component's `bound_functions.Inc` (nearest-history-entry lookup, which is what prompted the comment) have since been retired in favour of the `positionDiscrete`/`positionInterpolated` operator pair. Both `parameters/reference/evolutionGalaxyFormationNBody.xml:216` and `parameters/tutorials/lightconeOutput.xml:278` already use it.

Worth stating explicitly for the record: the choice of position operator is **not** optional for this purpose. With `positionDiscrete` the position is piecewise-constant across a whole evolution step (`discrete.F90:106-112` snaps to the nearest history entry at `differentialEvolutionStepFinalState`), so any orbit-dependent physics becomes a step function on ~0.1×t timescales. Only `positionInterpolated` makes the derived quantities usable.

**2. The ram-pressure symptom was routed around, not fixed.** `hotHaloRamPressureForceRelativePosition` (added in af2aa4d, Mar 2024, "useful for N-body merger trees where we have position information") computes the force from `position%position()`/`velocity()` of node and host rather than from the satellite component. It is what the N-body reference parameter file and `testSuite/parameters/subhaloOrbitOnZAxis.xml` actually use. So `hotHaloRamPressureForceOrbitalPosition` is still unusable with `preset`, but nothing in practice needs it to be.

### Assessment of the proposed solution

Making `preset` compute `position`/`velocity` from the node and host position components is, I think, still the right call — but the argument for it is now consolidation and unblocking, not ram pressure.

**In favour:**

- The mechanism already exists in exactly the right place. `preset` already carries two virtual, computed, gettable-only properties (`boundMass`, `nodeIndex`) implemented in `source/objects/nodes/components/satellite/preset/bound_functions.Inc`. Adding `position`/`velocity` the same way is a natural extension, and it preserves the established semantics (relative to the *immediate* host — see the accumulate-up-the-host-chain loop in `nodes/property_extractor/position_orbital.F90:108-120`).
- It fixes 26 consumers at one stroke. Today, `satellite%position()`/`velocity()` are read by 26 classes across tidal fields, tidal heating, tidal stripping, dynamical friction, SIDM evaporation/deceleration, sub-subhalo promotion, merging triggers, stream filters, lightcones and four property extractors. Exactly one class (`nodes/operators/physics/satellite_orbits.F90`) *owns* these properties; everything else is read-only and would just work. Continuing down the `relativePosition` road means a parallel implementation per consumer, which does not scale.
- It would make `hotHaloRamPressureForceRelativePosition` redundant, and that class currently builds, stores and destroys a `hotHaloMassDistribution_` that `relativePositionForce` never uses (`relative_position.F90:44,89,107` — the force uses `nodeHost%massDistribution(...)` instead). Consolidating would let that go.

**Against / caveats, in rough order of importance:**

- **The failure mode today is silent, not clean.** Only three call sites assert on satellite `position`/`velocity` (`hot_halo/ram_pressure_force/orbital_position.F90:81`, `nodes/operators/physics/subsubhalo_promotion.F90:74`, plus the position-history one in `satellite_status.F90`). Everywhere else, with `preset` active, the class-level default getter returns `nullDouble1d` — a **zero-size** array (`source/objects/nodes/_class.F90:222`). Assigning that into a `dimension(3)` local, or passing it to `Vector_Magnitude` (which declares `dimension(3)`, `math/vector.F90:48`), is a conformance violation: caught under bounds checking, undefined otherwise. Adding `componentPropertyAssert` to the remaining consumers is cheap, independent of everything else here, and worth doing regardless of which way this issue goes.
- **Orphans give zero force in the currently deployed configuration.** Once a subhalo's position history is exhausted, `positionInterpolated`'s trace switches the node to self-hosting (`interpolated.F90:641-646`), so its interpolated trajectory becomes *identical* to the host's — same knots, same cubic coefficients. `relativePositionForce` then computes `|pos − posHost| ≈ 0` and `|vel − velHost| ≈ 0`, so **orphans experience essentially no ram pressure**, when physically they should be the most strongly stripped population. Any implementation of the proposal inherits this unless it handles orphans explicitly. `nodeOperatorPositionOrphans` exists for this, but it does not compose with `positionInterpolated` — it extends `positionDiscrete` and writes at `differentialEvolutionStepFinalState`, so the next ODE RHS call overwrites it from the interpolation. Resolving that composition gap is arguably a prerequisite.
- **Three different "find the contemporaneous host" walks already exist**, because satellites can evolve ahead of their hosts: `relative_position.F90:134-160` (walk the host's descendant chain, pick the closest in time), `position_orbital.F90:113-131` (walk back down `firstChild` until times agree), and the host-tracking inside `interpolated.F90:612-629`. A virtual getter on `preset` needs the same logic and would be a fourth. This should be factored into one shared helper as part of the work, not reimplemented.
- **Hot-path cost.** These getters are read inside ODE right-hand sides. The existing `preset` virtual getters are not cheap — `SatellitePresetMergeBoundMass` constructs an `interpolator` object on *every* call. `position`/`velocity` would not need an interpolator (the position component is already up to date), so cost is a pointer chase, a host walk and a 3-vector subtract, but the host walk is unbounded in principle and should be measured.
- **Prerequisite on the position component.** The getter is only meaningful when `componentPosition` supports `position`/`velocity` and a position operator is active. With `componentPosition=null` it would return `nullDouble1d` again and reintroduce the same silent failure one level down. This needs its own assertion.
- **Small unknown:** every existing virtual gettable property in the codebase is rank 0. Rank-1 virtual getters with a user-supplied `getFunction` look supported by the generator (it just binds `procedure :: position => <userFunction>`, `Build/Components/Properties/Get.py:23-46`), but there is no precedent, so this needs a quick check before committing to the design.
- **Accuracy expectations should be set honestly.** The Merson et al. interpolation for satellites is a logarithmic spiral: radius varies as `exp(linear in t)` and angle linearly, in a fixed plane (`interpolated.F90:1017-1027`). It is monotonic in radius by construction, so a pericentre passage occurring between two snapshots is not reproduced — it is smoothed away. It was designed to produce plausible sky positions for lightcones, not to conserve orbital energy or angular momentum. For ram pressure and tidal stripping, which are dominated by pericentre, this systematically under-samples the peaks. This is a real physics limitation of the approach, not an implementation detail, and it argues for documenting the derived position/velocity on `preset` as "as good as the snapshot cadence allows" rather than as a drop-in equivalent of `orbiting`.

### Suggested shape of the work

1. Add `componentPropertyAssert` for satellite `position`/`velocity` to the remaining ~23 consumers that lack it. Independent, cheap, converts undefined behaviour into a clear error. Worth doing even if nothing else here proceeds.
2. Factor the "host node contemporaneous with this satellite" walk into one shared helper and retrofit the three existing copies.
3. Add virtual, gettable-only `position`/`velocity` to `preset` in `bound_functions.Inc`, using that helper, plus an assertion that the position component supports position/velocity. Gate on `presetOrbits`/`presetPositions` being available if an opt-out is wanted, though I would lean towards unconditional-and-asserted over a parameter.
4. Decide the orphan behaviour explicitly, and fix the `positionOrphans`/`positionInterpolated` composition gap. Leaving orphans at exactly the host centre is the current behaviour and it is wrong.
5. Once (3) lands, `hotHaloRamPressureForceOrbitalPosition` works with N-body trees and `relativePosition` becomes a candidate for deprecation.

`testSuite/parameters/subhaloOrbitOnZAxis.xml` (run in CI, `.github/workflows/cicd.yml:1673`) already exercises `preset` satellite + `positionInterpolated` + ram pressure and is the natural regression vehicle for all of this.
